google/resource_compute_region_instance_group_manager.go

```diff
@@ -0,0 +1,482 @@
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+)
+
+var RegionInstanceGroupManagerBaseApiVersion = v1
+var RegionInstanceGroupManagerVersionedFeatures = []Feature{Feature{Version: v0beta, Item: "auto_healing_policies"}}
+
+func resourceComputeRegionInstanceGroupManager() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceComputeRegionInstanceGroupManagerCreate,
+		Read:   resourceComputeRegionInstanceGroupManagerRead,
+		Update: resourceComputeRegionInstanceGroupManagerUpdate,
+		Delete: resourceComputeRegionInstanceGroupManagerDelete,
+		Exists: resourceComputeRegionInstanceGroupManagerExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"base_instance_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"instance_template": &schema.Schema{
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: compareSelfLinkRelativePaths,
+			},
+
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"region": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"fingerprint": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"instance_group": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"named_port": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"port": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			"self_link": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"target_pools": &schema.Schema{
+				Type:             schema.TypeSet,
+				Optional:         true,
+				DiffSuppressFunc: compareSelfLinkRelativePaths,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Set: selfLinkRelativePathHash,
+			},
+
+			"target_size": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+			},
+
+			"auto_healing_policies": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"health_check": &schema.Schema{
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: compareSelfLinkRelativePaths,
```

Comment:
DiffSuppressFunc works well on nested string. At least we use it in many places and I just did some manual testing and works as expected.

Reply:
Apologies, my information must be out of date. :) Glad it works! I remember something about the count getting off or something, but it must have been fixed since then.